### DISCUSSION TITLE
Choosing sigma_diff and radius parameters for SyN registration

### DISCUSSION THREAD

**Discussion by arokem** [dipy/dipy#2458] [Registration]
@aloofhawk asks (#2457) : 

Hello

I am using SyN to register two images. I dont have the pixel spacing information associated with the two images but they have been aligned using Rigid and Affine transform. I am running SyN now to further refine the registration. The images are quite similar but there are a lot of gyri and sulci present between the two images which dont align perfectly. But i can see that the algorithm gets close to finding a nice match. I wanted to know if there is a way to set the radius and sigma_diff paramters or some other parameter in the SyN algorithm to further finetune the registration. I can access the displacement field and I read somewhere in the code that the displacements are scaled by the spacing values. Are there any standard practices when choosing the above two parameters for the code.

**Comment by skoudoro** [comment]
Hi @aloofhawk,

Which metric are you using for the SyN? 

if you are using Cross-Correlation Similarity metric you can set them up using

```python
# Configure the metric
sigma_diff = 3.0
radius = 4
metric = metrics.CCMetric(2, sigma_diff, radius)
optimizer = imwarp.SymmetricDiffeomorphicRegistration(metric, level_iters)
optimizer.verbosity = VerbosityLevels.DEBUG
mapping = optimizer.optimize(static, moving, None)
m = optimizer.get_map()
```

I hope it helps and answers your question

**Reply by cakeinspace** [reply]
No i know how to set it but i want to know if there are any standard practices for choosing the best values
